### ISSUE TITLE
"go: add missing platforms from 'go tools dist list' (#2412)" not present in release-0.22 branch

### ISSUE THREAD

**Issue by BOTBrad** [bazelbuild/rules_go#2555]
### What version of rules_go are you using?

0.22.6

### What version of gazelle are you using?

0.21

### What version of Bazel are you using?

2.1.1

### Does this issue reproduce with the latest releases of all the above?

It doesn't happen in `go_rules` 0.21.7, and I'm fairly sure it won't happen in 0.23.3 .

### What operating system and processor architecture are you using?

Darwin x86_64

### Any other potentially useful information about your toolchain?

Not that I can think of for this issue.

### What did you do?

We're trying to upgrade to support go 1.14.4 from go 1.14.2. The minimal change was to change our `WORKSPACE` file to
use go 1.14.4 and update `go_rules` from 0.21.7 to 0.23.3 since 0.22.x doesn't seem to support 1.14.4. After this I tried to build a simple go_library bazel target.

### What did you expect to see?

Successful build.

### What did you see instead?

```
$ bazel build //<some dir>:go_default_library
ERROR: Analysis of target '//<some dir>:go_default_library' failed; build aborted: no such target '@io_bazel_rules_go//go/platform:illumos': target 'illumos' not declared in package 'go/platform' defined by /private/var/tmp/_bazel_cwywiorski/697e93d810e63190590fda4544dddf94/external/io_bazel_rules_go/go/platform/BUILD.bazel
INFO: Elapsed time: 0.217s
INFO: 0 processes.
FAILED: Build did NOT complete successfully (0 packages loaded, 1 target configured)
Bazel exited with status code 1
```

I think the issue is that [this change](https://github.com/bazelbuild/rules_go/pull/2412) was merged into 0.21.x and 0.23.x but not 0.22.x. I tried cloning 0.22.7 locally and adding `illumos` and `aix` entries to `GOOS_GOARCH` in `go/private/platforms.bzl` and everything built successfully. Looking at the history of that file, the missing platforms commit is present in release-0.21 and release-0.23 but not release-0.22. Upgrading to 0.23 would entail a full Bazel upgrade for us, so I'd like to stick with 0.22.x if possible. I'd be happy to start a PR of that change onto 0.22.x if that's the right move.

https://github.com/bazelbuild/rules_go/commits/release-0.21/go/private/platforms.bzl

https://github.com/bazelbuild/rules_go/commits/release-0.22/go/private/platforms.bzl

https://github.com/bazelbuild/rules_go/commits/release-0.23/go/private/platforms.bzl

**Comment by jayconrod** [comment]
This should have been picked onto `release-0.22`. I think it was just an oversight that it wasn't. I'll include it in the next release.

Please upgrade to v0.23.x if possible though. Go 1.14.4 is definitely supported. The `release-0.22` branch will only be maintained for another month or so.

**Comment by jayconrod** [comment]
I've cherry-picked #2412 and some other fixes onto `release-0.22`. I've been told there will be new Go point releases in a couple days, so I'll likely wait for that before tagging a rules_go release.
